src/test/java/com/softensy/SoftensySpringBoot/service/serviceImpl/DoctorServiceImplTest.java

```diff
@@ -0,0 +1,114 @@
+package com.softensy.SoftensySpringBoot.service.serviceImpl;
+
+import com.softensy.SoftensySpringBoot.entity.Doctor;
+import com.softensy.SoftensySpringBoot.repository.DoctorRepositoty;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.Mockito.*;
+
+
+@SpringBootTest
+class DoctorServiceImplTest {
+    @Autowired
+    private DoctorServiceImpl doctorService;
+
+    @MockBean
+    private DoctorRepositoty doctorRepositoty;
+
+    @Test
+    void givenDoctors_whenGetAllDoctors_thenReturnAllDoctors() {
+        List<Doctor> expectedDoctors = new ArrayList<>();
+        // given
+        Doctor firstDoctor = new Doctor("Ivan", "Ivanov", "Ivanovich", "Hirurg", Date.valueOf("1987-05-12"), 12345L);
+        firstDoctor.setId(1);
+        Doctor secondDoctor = new Doctor("Petr", "Petrov", "Petrov", "Terapevt", Date.valueOf("1988-07-19"), 54321L);
+        expectedDoctors.add(firstDoctor);
+        expectedDoctors.add(secondDoctor);
+        // when
+        when(doctorRepositoty.findAll()).thenReturn(expectedDoctors);
+        List<Doctor> actualDoctors = doctorService.getAllDoctors();
+        //then
+        Assertions.assertEquals(expectedDoctors, actualDoctors);
+        Assertions.assertNotNull(firstDoctor);
+        Assertions.assertNotNull(secondDoctor);
+        verify(doctorRepositoty).findAll();
+        verify(doctorRepositoty, times(1)).findAll();
+    }
+
+    @Test
+    void givenDoctor_whenGetDoctor_thenReturnDoctor() {
+        // given
+        Doctor expectedDoctor = new Doctor("Ivan", "Ivanov", "Ivanovich", "Hirurg", Date.valueOf("1987-05-12"), 12345);
+        expectedDoctor.setId(1l);
+        // when
+        when(doctorRepositoty.findById(1l)).thenReturn(Optional.of(expectedDoctor));
+        Doctor actualDoctor = doctorService.getDoctorById(1l).get();
```

Comment:
for long use plz capital L letter

https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.10.1

The suffix L is preferred, because the letter l (ell) is often hard to distinguish from the digit 1 (one).